programs/castle-vault/src/instructions/init.rs

```diff
@@ -194,7 +194,7 @@ pub fn handler(
     ctx.accounts.validate_referral_token()?;
 
     let vault = &mut ctx.accounts.vault;
-    vault.version = [2, 0, 0];
+    vault.version = [2, 0, 1];
```

Comment:
Unrelated to the PR, but may be helpful

```suggestion
    vault.version = [
        env!("CARGO_PKG_VERSION_MAJOR")
            .parse::<u8>()
            .expect("failed to parse major version"),
        env!("CARGO_PKG_VERSION_MINOR")
            .parse::<u8>()
            .expect("failed to parse minor version"),
        env!("CARGO_PKG_VERSION_PATCH")
            .parse::<u8>()
            .expect("failed to parse patch version"),
    ];
```